core/src/main/kotlin/fr/sncf/osrd/stdcm/preprocessing/implementation/BlockAvailability.kt

```diff
@@ -143,6 +147,13 @@ private fun convertWorkSchedules(
                 val chunkEndOffset = chunkStartOffset + infra.getTrackChunkLength(chunk).distance
                 if (chunkStartOffset > range.end || chunkEndOffset < range.begin) continue
                 val zone = infra.getTrackChunkZone(chunk)
+                if (zone == null) {
+                    // TODO: raise a warning
+                    blockAvailabilityLogger.warn(
+                        "Skipping work schedule because it is on a track not fully covered by routes: $track",
```

Comment:
I wonder if this really is a problem at that point: skipping occupation on something that can't be occupied anyway (so no wrong planning done using that resource), if I'm correct.
So that warning may be removed (or, as mentioned, replaced by some check at load on infra, but this is more work).
It may also just be downgraded to infos, because actually trying to use the zone may be a better sign of something off in data.

Then (if that log stay), I'd add detail to recover the affected work:
```suggestion
                    blockAvailabilityLogger.info(
                        "Skipping part of work schedule [${entry.startTime}; ${entry.endTime}] because it is on a track not fully covered by routes: $track",
```

Reply:
You are correct, it can't be occupied anyway, since there are no route for this chunk. However, I'd still prefer to raise a warning, because as a user, I'd rather know why my work schedule got plain out ignored when I tried inputting them without having to search through the code :)
I like your commit suggestion though, it adds more details.